pinecone/index_connection_test.go

```diff
@@ -4,114 +4,26 @@ import (
+	"google.golang.org/protobuf/types/known/structpb"
 
-	"github.com/google/uuid"
-	"github.com/pinecone-io/go-pinecone/internal/utils"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"github.com/stretchr/testify/suite"
-	"google.golang.org/grpc"
-	"google.golang.org/protobuf/types/known/structpb"
 )
 
-// Integration tests:
-type IndexConnectionTestsIntegration struct {
-	suite.Suite
-	host             string
-	dimension        int32
-	apiKey           string
-	indexType        string
-	idxConn          *IndexConnection
-	sourceTag        string
-	idxConnSourceTag *IndexConnection
-	vectorIds        []string
-}
-
-func TestIntegrationIndexConnection(t *testing.T) {
-	apiKey := os.Getenv("PINECONE_API_KEY")
-	assert.NotEmptyf(t, apiKey, "PINECONE_API_KEY env variable not set")
-
-	client, err := NewClient(NewClientParams{ApiKey: apiKey})
-	if err != nil {
-		t.FailNow()
-	}
-
-	podIndexName := os.Getenv("TEST_PODS_INDEX_NAME")
-	assert.NotEmptyf(t, podIndexName, "TEST_PODS_INDEX_NAME env variable not set")
-
-	podIdx, err := client.DescribeIndex(context.Background(), podIndexName)
-	if err != nil {
-		t.FailNow()
-	}
-
-	podTestSuite := new(IndexConnectionTestsIntegration)
-	podTestSuite.indexType = "pod"
-	podTestSuite.host = podIdx.Host
-	podTestSuite.dimension = podIdx.Dimension
-	podTestSuite.apiKey = apiKey
-
-	serverlessIndexName := os.Getenv("TEST_SERVERLESS_INDEX_NAME")
-	assert.NotEmptyf(t, serverlessIndexName, "TEST_SERVERLESS_INDEX_NAME env variable not set")
-
-	serverlessIdx, err := client.DescribeIndex(context.Background(), serverlessIndexName)
-	if err != nil {
-		t.FailNow()
-	}
-
-	serverlessTestSuite := new(IndexConnectionTestsIntegration)
-	serverlessTestSuite.indexType = "serverless"
-	serverlessTestSuite.host = serverlessIdx.Host
-	serverlessTestSuite.dimension = serverlessIdx.Dimension
-	serverlessTestSuite.apiKey = apiKey
-
-	suite.Run(t, podTestSuite)
-	suite.Run(t, serverlessTestSuite)
-}
-
-func (ts *IndexConnectionTestsIntegration) SetupSuite() {
-	assert.NotEmptyf(ts.T(), ts.host, "HOST env variable not set")
-	assert.NotEmptyf(ts.T(), ts.apiKey, "API_KEY env variable not set")
-	additionalMetadata := map[string]string{"api-key": ts.apiKey}
-
-	namespace, err := uuid.NewV7()
-	assert.NoError(ts.T(), err)
-
-	idxConn, err := newIndexConnection(newIndexParameters{
-		additionalMetadata: additionalMetadata,
-		host:               ts.host,
-		namespace:          namespace.String(),
-		sourceTag:          ""})
-	assert.NoError(ts.T(), err)
-	ts.idxConn = idxConn
-
-	ts.sourceTag = "test_source_tag"
-	idxConnSourceTag, err := newIndexConnection(newIndexParameters{
-		additionalMetadata: additionalMetadata,
-		host:               ts.host,
-		namespace:          namespace.String(),
-		sourceTag:          ts.sourceTag})
-	assert.NoError(ts.T(), err)
-	ts.idxConnSourceTag = idxConnSourceTag
-
-	ts.loadData()
+func TestIndexConnectionIntegration(t *testing.T) {
+	RunSuites(t)
```

Comment:
So we call `RunSuites()` from here in each test file (`client_test.go` and `index_connection_test.go`). Then in `RunSuites()` we're creating two `IntegrationTest` structs and then calling `suite.Run` on both of them:

```go
	suite.Run(t, podTestSuite)
	suite.Run(t, serverlessTestSuite)
```

This may be a testify thing I'm not clear on, but is there a possibility we're running duplicates of the test suites? Like if we trigger `go test` on both files, does it spawn different instances of `testing.T` and `RunSuites()` etc?

Reply:
Gone! Now we have `RunSuites` in `suite_runner_test.go` (has to be append with `_test` so that `go test` runs the `RunSuites` func, and `test_suite.go`, which is where the setup/teardown lives.